File: a-collections-benchmark/src/main/java/com/ajjpj/acollections/immutable/j/Tree.java
package com.ajjpj.acollections.immutable.j;


public class Tree<A, B> {
    private final A key;
    private final B value;
    private final Tree<A, B> left, right;

    public Tree (A key, B value, Tree<A, B> left, Tree<A, B> right) {
        this.key = key;
        this.value = value;
        this.left = left;
        this.right = right;
    }

    public A key() {
        return key;
    }
    public B value() {
        return value;
    }
    public Tree<A,B> left() {
        return left;
    }
    public Tree<A,B> right() {
        return right;
    }
}

File: a-collections-benchmark/src/main/scala/com/ajjpj/acollections/immutable/s/Tree.scala
package com.ajjpj.acollections.immutable.s

import scala.annotation.meta.getter


class Tree[A, +B](@(inline @getter) final val key: A,
                  @(inline @getter) final val value: B,
                  @(inline @getter) final val left: Tree[A, B],
                  @(inline @getter) final val right: Tree[A, B])
